src/components/liveSession/services/liveSession.service.js

```diff
@@ -249,24 +245,43 @@
             };
 
             this._updateLesson = (liveSessionData) => {
+                function updateSingleLesson(lesson) {
+                    // update lesson startTime, endTime and status
+                    lesson.startTime = lesson.startTime ? lesson.startTime : liveSessionData.startTime ? liveSessionData.startTime: null;
+                    lesson.endTime = lesson.endTime? lesson.data.endTime : liveSessionData.endTime ? liveSessionData.endTime : null;
+                    lesson.status = lesson.status === LessonStatusEnum.SCHEDULED.enum ?
+                        LessonStatusEnum.ATTENDED.enum : lesson.status;
+                    lesson.lessonNotes = lesson.lessonNotes || {};
+                    lesson.lessonNotes.status = lesson.lessonNotes.status || LessonNotesStatusEnum.PENDING_NOTES.enum;
+                    lesson.liveSessions = lesson.liveSessions || [];
+                    lesson.liveSessions.push(liveSessionData.guid);
+                    lesson.liveSessions = UtilitySrv.array.removeDuplicates(lesson.liveSessions);
+                    return ZnkLessonNotesSrv.updateLesson(lesson);
+                }
                 return LiveSessionUiSrv.isDarkFeaturesValid(liveSessionData.educatorId, liveSessionData.studentId)
                     .then(isDarkFeaturesValid => {
                         if (isDarkFeaturesValid) {
-                            return ZnkLessonNotesSrv.getLessonById(liveSessionData.lessonId).then(lesson => {
-                                if (lesson.data.id) {
-                                    // update lesson startTime, endTime and status
-                                    lesson.data.startTime = lesson.data.startTime ? lesson.data.startTime : liveSessionData.startTime ? liveSessionData.startTime: null;
-                                    lesson.data.endTime = lesson.data.endTime? lesson.data.endTime : liveSessionData.endTime ? liveSessionData.endTime : null;
-                                    lesson.data.status = LessonStatusEnum.ATTENDED.enum;
-                                    lesson.data.lessonNotes = lesson.data.lessonNotes || {};
-                                    lesson.data.lessonNotes.status = lesson.data.lessonNotes.status || LessonNotesStatusEnum.PENDING_NOTES.enum;
-                                    try {
-                                        return ZnkLessonNotesSrv.updateLesson(lesson.data).then(updatedLesson => {
-                                            $log.debug('_updateLesson: update lesson startTime & status. updatedLesson: ', updatedLesson);
-                                        });
-                                    } catch (err) {
-                                        $log.error('_updateLesson: updateLesson failed. Error: ', err);
+                            return ZnkLessonNotesSrv.getLessonById(liveSessionData.lessonId).then(lessonData => {
```

Comment:
Instead of getting the lesson and then the other back2back lessons, save the backToBackId in the session obj when creating the session.

Then, in this func, check if session.backToBackId exists, if so then get the back2back lessons in one query, otherwise get the single lesson by session.lessonId.